server/src/test/java/org/elasticsearch/env/NodeMetadataTests.java

```diff
@@ -34,22 +34,34 @@ private Version randomVersion() {
 
     public void testEqualsHashcodeSerialization() {
         final Path tempDir = createTempDir();
-        EqualsHashCodeTestUtils.checkEqualsAndHashCode(new NodeMetadata(randomAlphaOfLength(10), randomVersion()), nodeMetadata -> {
-            final long generation = NodeMetadata.FORMAT.writeAndCleanup(nodeMetadata, tempDir);
-            final Tuple<NodeMetadata, Long> nodeMetadataLongTuple = NodeMetadata.FORMAT.loadLatestStateWithGeneration(
-                logger,
-                xContentRegistry(),
-                tempDir
-            );
-            assertThat(nodeMetadataLongTuple.v2(), equalTo(generation));
-            return nodeMetadataLongTuple.v1();
-        }, nodeMetadata -> {
-            if (randomBoolean()) {
-                return new NodeMetadata(randomAlphaOfLength(21 - nodeMetadata.nodeId().length()), nodeMetadata.nodeVersion());
-            } else {
-                return new NodeMetadata(nodeMetadata.nodeId(), randomValueOtherThan(nodeMetadata.nodeVersion(), this::randomVersion));
+        EqualsHashCodeTestUtils.checkEqualsAndHashCode(
+            new NodeMetadata(randomAlphaOfLength(10), randomVersion(), randomVersion()),
+            nodeMetadata -> {
+                final long generation = NodeMetadata.FORMAT.writeAndCleanup(nodeMetadata, tempDir);
+                final Tuple<NodeMetadata, Long> nodeMetadataLongTuple = NodeMetadata.FORMAT.loadLatestStateWithGeneration(
+                    logger,
+                    xContentRegistry(),
+                    tempDir
+                );
+                assertThat(nodeMetadataLongTuple.v2(), equalTo(generation));
+                return nodeMetadataLongTuple.v1();
+            },
+            nodeMetadata -> {
+                if (randomBoolean()) {
```

Comment:
Could we have a third branch which mutates the new field and leaves the other two fields alone, to verify that `equals()` does compare the new field too?